src/components/actions/coreActions.ts

```diff
@@ -0,0 +1,493 @@
+      sequence: 100,
+      condition: true,
+      parameter_queries: {
+        pattern: '<?templateName> - <?serial>',
+        fieldName: 'name',
+        tableName: 'application',
+        customFields: {
+          serial: 'applicationData.applicationSerial',
+          templateName: 'applicationData.templateName',
+        },
+        updateRecord: true,
+      },
+    },
+  ],
+  [Trigger.OnApplicationRestart]: [
+    // Set status = DRAFT
+    {
+      code: 'changeStatus',
+      path: '../plugins/action_change_status/src/index.ts',
+      name: 'Change Status',
+      trigger: 'ON_APPLICATION_SUBMIT',
+      event_code: null,
+      sequence: -1,
+      condition: true,
+      parameter_queries: { newStatus: 'SUBMITTED' },
+    },
+  ],
+  [Trigger.OnApplicationSubmit]: [
+    // Set status = SUBMITTED
+    {
+      code: 'changeStatus',
+      path: '../plugins/action_change_status/src/index.ts',
+      name: 'Change Status',
+      trigger: 'ON_APPLICATION_SUBMIT',
+      event_code: null,
+      sequence: -6,
+      condition: true,
+      parameter_queries: { newStatus: 'SUBMITTED' },
+    },
+    // Trim application responses that haven't changed since previous submission
+    {
+      code: 'trimResponses',
+      path: '../plugins/action_trim_responses/src/index.ts',
+      name: 'Trim duplicate responses',
+      trigger: 'ON_APPLICATION_SUBMIT',
+      event_code: null,
+      sequence: -5,
+      condition: true,
+      parameter_queries: {},
+    },
+    // Create review assignment records for the next review level
+    {
+      code: 'generateReviewAssignments',
+      path: '../plugins/action_generate_review_assignment_records/src/index.ts',
+      name: 'Generate Review Assignment Records',
+      trigger: 'ON_APPLICATION_SUBMIT',
+      event_code: null,
+      sequence: -4,
+      condition: true,
+      parameter_queries: {},
+    },
+    // Remove any files that were uploaded but not submitted as part of the
+    // application
+    {
+      code: 'cleanupFiles',
+      path: '../plugins/action_files_cleanup/src/index.ts',
+      name: 'Clean up application files',
+      trigger: 'ON_APPLICATION_SUBMIT',
+      event_code: null,
+      sequence: -3,
+      condition: true,
+      parameter_queries: {},
+    },
+    // Update review statuses
+    // TO-DO: Un-comment once action is re-implemented
+    // {
+    //   code: 'updateReviewStatuses',
+    //   path: '../plugins/action_update_review_statuses/src/index.ts',
+    //   name: 'Update Review Statuses',
+    //   trigger: 'ON_APPLICATION_SUBMIT',
+    //   event_code: null,
+    //   sequence: -2,
+    //   condition: true,
+    //   parameter_queries: {
+    //     changedResponses: {
+    //       operator: 'objectProperties',
+    //       children: ['outputCumulative.updatedResponses'],
+    //     },
+    //   },
+    // },
+
+    // Change outcome to APPROVED if there are no other "changeOutcome" actions
+    // associated with this template
```

Comment:
```suggestion
    // associated with this template. 
    // Used for Non-reviewable templates or when there isn't another specific changeOutcome included
```

Reply:
I wonder how would this behave when you "create a template" (pretend we don't do copies only) and this changeOutcome is included first, but the user goes and add a new specific one.

Reply:
I think it would be fine. This action wouldn't run, since it'd find another one.